src/content/blog/try-catch-as-little-as-possible/index.mdx

```diff
@@ -0,0 +1,238 @@
+}
+```
+
+`fileIncludes` will throw if no file exists under `filePath`.
+So, it might be tempting to wrap the function's body in a `try...catch`:
+
+```ts
+import fs from "node:fs/promises";
+
+async function fileIncludes(filePath: string, search: string) {
+	// 🛑 See later - there's a better way!
+	try {
+		const data = await fs.readFile(filePath);
+		const text = data.toString();
+
+		return text.includes(search);
+	} catch (error) {
+		console.warn(`Failed to read ${filePath}:`, error);
+		return false;
+	}
+}
+```
+
+There's no need to wrap _all_ of the code inside `fileIncludes` in the `try...catch`.
+Only the call to `fs.readFile` is expected to throw.
+Even if another line of code were to throw, it wouldn't likely be for the logged reason _"Failed to read ..."_.
+
+A more targeted strategy would be to only wrap the one line in a `try...catch`:
+
+```ts
+import fs from "node:fs/promises";
+
+async function fileIncludes(filePath: string, search: string) {
+	let data;
+
+	try {
+		data = await fs.readFile(filePath);
+	} catch (error) {
+		console.warn(`Failed to read ${filePath}:`, error);
+		return false;
+	}
+
+	const text = data.toString();
+
+	return text.includes(search);
+}
+```
+
+This way, only the one line that might throw an error is handled by the `try...catch`.
+Developers reading the `fileIncludes` function can more easily understand what specifically might fail.
+Any other unexpected error that occurs would fall back to however the project handles unexpected errors.
+
+<LabeledVideo
+	alt="Ice Cube in 'Check Yo Self' saying 'check yourself before you wreck yourself'."
+	description="Ice Cube's been saying this for years. Remembering to check is an essential step for avoiding a wreck."
+	original="https://tenor.com/view/wreck-check-yourself-before-you-wreck-yourself-check-yo-self-ice-cube-gif-5076649"
+	src={checkYourselfBeforeYouWreckYourselfIce}
+/>
+
+## Further Reading
+
+If you want to read more about why `try...catch` is something to avoid when possible, I recommend:
+
+- [Exceptions as a form of control flow are inherently difficult to reason about](https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why)
+- [Thrown errors can't be modeled by type systems like TypeScript's](https://www.learningtypescript.com/articles/why-typescript-doesnt-include-a-throws-keyword)
+
+## Don't be afraid to ask for help
+
+It's ok if you're having a hard time figuring out the right way to safely handle errors in code!
+I want open source software to be a warm, welcoming place.
+Nobody should feel intimidated asking me for help with code.
+
+Go ahead and post a comment explaining what you've tried, what did or didn't work, and any specific questions you have.
+I'll be happy to give you pointers and work with you to write the code safely.
+
+Thank you for sending a pull request in, let alone working with me on getting it approved! 💙
```

Comment:
As much as I want to include some welcoming note... I suspect these blog posts could be useful to other people. I'll probably remove this in favor of staying welcoming in my PR reviews.